src/plugins/intel_gpu/src/graph/slice.cpp

```diff
@@ -10,6 +10,30 @@
 #include <json_object.h>
 
 namespace cldnn {
+
+SliceKernelRefNeededInputs SliceKernelRefNeededInputs::Create(const slice_node& node) {
+    SliceKernelRefNeededInputs inputs;
+
+    const auto& node_inputs = node.get_dependencies();
+
+    const bool axes_in_runtime =
+        ((node_inputs.size() == InputIndices::kInputsNum) && !node_inputs[InputIndices::kAxes].first->is_constant());
+    const bool start_in_runtime = node_inputs[InputIndices::kStart].first->is_dynamic();
+    const bool step_in_runtime = node_inputs[InputIndices::kStep].first->is_dynamic();
```

Comment:
This `is_dynamic()` method checks if program_node has any dynamic input/outputs _in terms of shape_ not the data, so if your intention was to check the data itself (if it's constant data or not), you need to use `!is_constant()`